src/main/scala/com/microsoft/partnercatalyst/fortis/spark/StreamProviderFactory.scala

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+import java.io.File
+
+import com.microsoft.partnercatalyst.fortis.spark.streamfactories.adapters.TadawebAdapter
+import com.microsoft.partnercatalyst.fortis.spark.streamfactories._
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.StreamProvider
+
+object StreamProviderFactory {
+  def create()(implicit settings: Settings): StreamProvider = {
```

Comment:
For Settings specifically, it's intended to be somewhat of a global variable. I think that makes it a great candidate for implicit. We can treat it as such without actually coupling our code to a singleton.